extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt

```diff
@@ -33,4 +33,4 @@ class VertxDispatcher(private val vertxContext: Context) : CoroutineDispatcher()
             block.run()
```

Comment:
shouldn't this be checking that we are on the same context as the one passed above?

Reply:
Good point. I didn't really change this, this what is what we already have

Reply:
@cescoffier I am not sure if I understand you correctly, but are you aware of this [discussion](https://github.com/quarkusio/quarkus/pull/16931#discussion_r624789718)?

Reply:
Yes, I've seen it. I'm not totally convinced by the "do not call this method immediately". In general, it's a good optimization avoiding going back to the waiting queue. I can understand why they want to enforce this, but in Vert.x typically, we often decide to call immediately if we are already on the right context for performance reasons.